src/pip/_internal/req/req_install.py

```diff
@@ -571,8 +571,30 @@ def get_pep_518_info(self):
         if os.path.isfile(self.pyproject_toml):
             with io.open(self.pyproject_toml, encoding="utf-8") as f:
                 pp_toml = pytoml.load(f)
-            build_sys = pp_toml.get('build-system', {})
-            return (build_sys.get('requires', ['setuptools', 'wheel']), True)
+
+            build_system = pp_toml.get('build-system', {})
+            if "requires" not in build_system:
+                raise InstallationError(
+                    "{} does not comply with PEP 518 as it since it's "
```

Comment:
`as it since it's`?

Reply:
Also, the function could be simplified to return the list of build requirements or `None` if PEP 518 support is not used, no need for the `isolate` return value anymore (and the corresponding code in `prep_for_dist` simplified too).

Reply:
Indeed. Let's simplify the message grouping logic.